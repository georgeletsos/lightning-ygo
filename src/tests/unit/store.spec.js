import axios from "axios";
import Vuex from "vuex";
import { cloneDeep } from "lodash-es";
import { createLocalVue } from "@vue/test-utils";
import cardsModule from "@/store/cards.module";
import {
  FETCH_CATALOG_CARDS,
  FETCH_DISPLAY_CARDS,
  FETCH_PREV_DISPLAY_CARD,
  FETCH_NEXT_DISPLAY_CARD,
  RESET_CATALOG_CARDS,
  RESET_DISPLAY_CARDS,
  CHANGE_DISPLAY_CARD
} from "@/store/actions.type";
import { mockCards } from "./mocks";

jest.mock("axios");
axios.get.mockResolvedValue({ data: mockCards });

const localVue = createLocalVue();
localVue.use(Vuex);

describe("`cards.module`", () => {
  let store;
  beforeEach(() => {
    store = new Vuex.Store(cloneDeep(cardsModule));
  });

  it("FETCH_CATALOG_CARDS - should fetch `state.catalogCards` from the API", async () => {
    expect.assertions(1);

    await store.dispatch(FETCH_CATALOG_CARDS, {});
    expect(store.state.catalogCards).not.toHaveLength(0);
  });

  it("FETCH_DISPLAY_CARDS - should fetch any Main Deck Monsters as `state.displayCards`", () => {
    store.state.catalogCards = cloneDeep(mockCards);
    const displayFilters = [
      {
        active: true,
        cardType: "monster",
        excludedTypes: ["fusion", "synchro"]
      }
    ];

    store.dispatch(FETCH_DISPLAY_CARDS, displayFilters);
    expect(store.state.displayCards).toMatchSnapshot();
  });

  it("FETCH_DISPLAY_CARDS - should fetch any Extra Deck Monsters as `state.displayCards`", () => {
    store.state.catalogCards = cloneDeep(mockCards);
    const displayFilters = [
      {
        active: true,
        cardType: "monster",
        includedTypes: ["fusion", "synchro"]
      }
    ];

    store.dispatch(FETCH_DISPLAY_CARDS, displayFilters);
    expect(store.state.displayCards).toMatchSnapshot();
  });

  it("FETCH_DISPLAY_CARDS - should fetch any Spells/Traps as `state.displayCards`", () => {
    store.state.catalogCards = cloneDeep(mockCards);
    const displayFilters = [
      {
        active: true,
        cardType: "spell"
      },
      {
        active: true,
        cardType: "trap"
      }
    ];

    store.dispatch(FETCH_DISPLAY_CARDS, displayFilters);
    expect(store.state.displayCards).toMatchSnapshot();
  });

  it("FETCH_PREV_DISPLAY_CARD - should change `state.displayCard` to the previous card of `state.displayCards`", () => {
    store.state.displayCards = cloneDeep(mockCards);
    const mockCardIndex = 2;
    const mockCard = mockCards[mockCardIndex];

    store.dispatch(FETCH_PREV_DISPLAY_CARD, mockCard);
    expect(store.state.displayCard).toEqual(mockCards[mockCardIndex - 1]);
  });

  it("FETCH_PREV_DISPLAY_CARD - when the current `state.displayCard` is the first card; should change `state.displayCard` to the last card of `state.displayCards`", () => {
    store.state.displayCards = cloneDeep(mockCards);
    const mockCard = mockCards[0];

    store.dispatch(FETCH_PREV_DISPLAY_CARD, mockCard);
    expect(store.state.displayCard).toEqual(mockCards[mockCards.length - 1]);
  });

  it("FETCH_NEXT_DISPLAY_CARD - should change `state.displayCard` to the next card of `state.displayCards`", () => {
    store.state.displayCards = cloneDeep(mockCards);
    const mockCardIndex = 2;
    const mockCard = mockCards[mockCardIndex];

    store.dispatch(FETCH_NEXT_DISPLAY_CARD, mockCard);
    expect(store.state.displayCard).toEqual(mockCards[mockCardIndex + 1]);
  });

  it("FETCH_NEXT_DISPLAY_CARD - when the current `state.displayCard` is the last card; should change `state.displayCard` to the first card of `state.displayCards`", () => {
    store.state.displayCards = cloneDeep(mockCards);
    const mockCard = mockCards[mockCards.length - 1];

    store.dispatch(FETCH_NEXT_DISPLAY_CARD, mockCard);
    expect(store.state.displayCard).toEqual(mockCards[0]);
  });

  it("RESET_CATALOG_CARDS - should reset `state.catalogCards`", () => {
    store.state.catalogCards = cloneDeep(mockCards);

    store.dispatch(RESET_CATALOG_CARDS);
    expect(store.state.catalogCards).toHaveLength(0);
  });

  it("RESET_DISPLAY_CARDS - should reset `state.displayCards`", () => {
    store.state.displayCards = cloneDeep(mockCards);

    store.dispatch(RESET_DISPLAY_CARDS);
    expect(store.state.displayCards).toHaveLength(0);
  });

  it("CHANGE_DISPLAY_CARD - should change `state.displayCard`", () => {
    const mockCard = mockCards[0];
    store.dispatch(CHANGE_DISPLAY_CARD, mockCard);
    expect(store.state.displayCard).toEqual(mockCard);
  });
});
