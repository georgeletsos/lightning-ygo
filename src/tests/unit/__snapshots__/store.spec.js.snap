// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`|cards.module| should fetch any Extra Deck Monsters as |state.displayCards| 1`] = `
Array [
  Object {
    "atk": "?",
    "attribute": "dark",
    "cardType": "monster",
    "def": "?",
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279252/lightning_ygo_api/card_images/art/64599569.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279250/lightning_ygo_api/card_images/big/64599569.jpg",
      "id": "64599569",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279251/lightning_ygo_api/card_images/small/64599569.jpg",
    },
    "level": 9,
    "monsterType": "machine",
    "name": "Chimeratech Overdragon",
    "text": "\\"Cyber Dragon\\" + 1+ Machine monsters
Must be Fusion Summoned. If this card is Fusion Summoned: Send all other cards you control to the GY. The original ATK/DEF of this card each become equal to the number of Fusion Materials used for its Fusion Summon x 800. Each turn, this card can attack your opponent's monsters a number of times equal to the number of Fusion Materials used for its Fusion Summon.",
    "types": Array [
      "fusion",
      "effect",
    ],
  },
  Object {
    "atk": 2100,
    "attribute": "dark",
    "cardType": "monster",
    "def": 1600,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942047/lightning_ygo_api/card_images/art/84058253.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942045/lightning_ygo_api/card_images/big/84058253.jpg",
      "id": "84058253",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942046/lightning_ygo_api/card_images/small/84058253.jpg",
    },
    "level": 5,
    "monsterType": "machine",
    "name": "Chimeratech Rampage Dragon",
    "text": "2+ \\"Cyber Dragon\\" monsters
A Fusion Summon of this card can only be done with the above Fusion Materials. When this card is Fusion Summoned: You can target Spells/Traps on the field, up to the number of Fusion Materials used for its Fusion Summon; destroy them. Once per turn: You can send up to 2 LIGHT Machine monsters from your Deck to the GY, and if you do, for each monster sent to the GY, this card gains 1 additional attack during each Battle Phase this turn.",
    "types": Array [
      "fusion",
      "effect",
    ],
  },
  Object {
    "atk": 4000,
    "attribute": "light",
    "cardType": "monster",
    "def": 2800,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279992/lightning_ygo_api/card_images/art/1546123.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279989/lightning_ygo_api/card_images/big/1546123.jpg",
      "id": "1546123",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279991/lightning_ygo_api/card_images/small/1546123.jpg",
    },
    "level": 10,
    "monsterType": "machine",
    "name": "Cyber End Dragon",
    "text": "\\"Cyber Dragon\\" + \\"Cyber Dragon\\" + \\"Cyber Dragon\\"
A Fusion Summon of this card can only be done with the above Fusion Materials. If this card attacks a Defense Position monster, inflict piercing battle damage.",
    "types": Array [
      "fusion",
      "effect",
    ],
  },
  Object {
    "atk": 2800,
    "attribute": "light",
    "cardType": "monster",
    "def": 2100,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280086/lightning_ygo_api/card_images/art/74157028.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280084/lightning_ygo_api/card_images/big/74157028.jpg",
      "id": "74157028",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280085/lightning_ygo_api/card_images/small/74157028.jpg",
    },
    "level": 8,
    "monsterType": "machine",
    "name": "Cyber Twin Dragon",
    "text": "\\"Cyber Dragon\\" + \\"Cyber Dragon\\"
A Fusion Summon of this card can only be done with the above Fusion Materials. This card can make a second attack during each Battle Phase.",
    "types": Array [
      "fusion",
      "effect",
    ],
  },
]
`;

exports[`|cards.module| should fetch any Main Deck Monsters as |state.displayCards| 1`] = `
Array [
  Object {
    "atk": 0,
    "attribute": "wind",
    "cardType": "monster",
    "def": 2000,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1586941744/lightning_ygo_api/card_images/art/67159705.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1586941742/lightning_ygo_api/card_images/big/67159705.jpg",
      "id": "67159705",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1586941743/lightning_ygo_api/card_images/small/67159705.jpg",
    },
    "level": 4,
    "monsterType": "machine",
    "name": "Armored Cybern",
    "text": "Once per turn, you can either: Target 1 \\"Cyber Dragon\\" you control, or 1 Fusion Monster you control that lists \\"Cyber Dragon\\" as Fusion Material; equip this card to that target, OR: Unequip this card and Special Summon it. Once per turn, while equipped to a monster by this effect: You can target 1 face-up monster on the field; the equipped monster loses exactly 1000 ATK (even if this card leaves the field), and if it does, destroy that target. If the equipped monster would be destroyed by battle or card effect, destroy this card instead.",
    "types": Array [
      "union",
      "effect",
    ],
  },
  Object {
    "atk": 2100,
    "attribute": "light",
    "cardType": "monster",
    "def": 1600,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279976/lightning_ygo_api/card_images/art/70095154.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279974/lightning_ygo_api/card_images/big/70095154.jpg",
      "id": "70095154",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279975/lightning_ygo_api/card_images/small/70095154.jpg",
    },
    "level": 5,
    "monsterType": "machine",
    "name": "Cyber Dragon",
    "text": "If only your opponent controls a monster, you can Special Summon this card (from your hand).",
    "types": Array [
      "effect",
    ],
  },
  Object {
    "atk": 400,
    "attribute": "light",
    "cardType": "monster",
    "def": 1500,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279980/lightning_ygo_api/card_images/art/23893227.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279977/lightning_ygo_api/card_images/big/23893227.jpg",
      "id": "23893227",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279978/lightning_ygo_api/card_images/small/23893227.jpg",
    },
    "level": 2,
    "monsterType": "machine",
    "name": "Cyber Dragon Core",
    "text": "When this card is Normal Summoned: Add 1 \\"Cyber\\" Spell/Trap from your Deck to your hand. If only your opponent controls a monster: You can banish this card from your GY; Special Summon 1 \\"Cyber Dragon\\" monster from your Deck. You can only use 1 \\"Cyber Dragon Core\\" effect per turn, and only once that turn. This card's name becomes \\"Cyber Dragon\\" while on the field or in the GY.",
    "types": Array [
      "effect",
    ],
  },
  Object {
    "atk": 1800,
    "attribute": "light",
    "cardType": "monster",
    "def": 800,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279983/lightning_ygo_api/card_images/art/59281922.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279981/lightning_ygo_api/card_images/big/59281922.jpg",
      "id": "59281922",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279982/lightning_ygo_api/card_images/small/59281922.jpg",
    },
    "level": 4,
    "monsterType": "machine",
    "name": "Cyber Dragon Drei",
    "text": "When this card is Normal Summoned: You can make all \\"Cyber Dragon\\" you currently control become Level 5. You cannot Special Summon any monsters during the turn you activate this effect, except Machine monsters. If this card is banished: You can target 1 \\"Cyber Dragon\\" you control; it cannot be destroyed by battle or card effects this turn. This card's name becomes \\"Cyber Dragon\\" while on the field or in the GY.",
    "types": Array [
      "effect",
    ],
  },
  Object {
    "atk": 1100,
    "attribute": "light",
    "cardType": "monster",
    "def": 1600,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942144/lightning_ygo_api/card_images/art/29975188.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942142/lightning_ygo_api/card_images/big/29975188.jpg",
      "id": "29975188",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942143/lightning_ygo_api/card_images/small/29975188.jpg",
    },
    "level": 4,
    "monsterType": "machine",
    "name": "Cyber Dragon Vier",
    "text": "This card's name becomes \\"Cyber Dragon\\" while on the field or in the GY. If you Normal or Special Summon \\"Cyber Dragon\\", except during the Damage Step: You can Special Summon this card from your hand in Defense Position. You can only use this effect of \\"Cyber Dragon Vier\\" once per turn. Each \\"Cyber Dragon\\" you control gains 500 ATK/DEF.",
    "types": Array [
      "effect",
    ],
  },
  Object {
    "atk": 1500,
    "attribute": "light",
    "cardType": "monster",
    "def": 1000,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279988/lightning_ygo_api/card_images/art/5373478.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279984/lightning_ygo_api/card_images/big/5373478.jpg",
      "id": "5373478",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584279986/lightning_ygo_api/card_images/small/5373478.jpg",
    },
    "level": 4,
    "monsterType": "machine",
    "name": "Cyber Dragon Zwei",
    "text": "If this card attacks an opponent's monster, it gains 300 ATK during the Damage Step only. Once per turn: You can reveal 1 Spell in your hand; this card's name becomes \\"Cyber Dragon\\" until the End Phase. This card's name becomes \\"Cyber Dragon\\" while in the GY.",
    "types": Array [
      "effect",
    ],
  },
  Object {
    "atk": 1100,
    "attribute": "light",
    "cardType": "monster",
    "def": 600,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584290785/lightning_ygo_api/card_images/art/26439287.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584290782/lightning_ygo_api/card_images/big/26439287.jpg",
      "id": "26439287",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584290784/lightning_ygo_api/card_images/small/26439287.jpg",
    },
    "level": 3,
    "monsterType": "machine",
    "name": "Proto-Cyber Dragon",
    "text": "This card's name becomes \\"Cyber Dragon\\" while on the field.",
    "types": Array [
      "effect",
    ],
  },
  Object {
    "atk": 2100,
    "attribute": "light",
    "cardType": "monster",
    "def": 1600,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584295044/lightning_ygo_api/card_images/art/83629030.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584295042/lightning_ygo_api/card_images/big/83629030.jpg",
      "id": "83629030",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584295043/lightning_ygo_api/card_images/small/83629030.jpg",
    },
    "level": 5,
    "monsterType": "machine",
    "name": "Toon Cyber Dragon",
    "text": "If your opponent controls a monster and you control no monsters, you can Special Summon this card (from your hand). Cannot attack the turn it is Summoned. While you control \\"Toon World\\" and your opponent controls no Toon monsters, this card can attack your opponent directly.",
    "types": Array [
      "toon",
      "effect",
    ],
  },
]
`;

exports[`|cards.module| should fetch any Spells/Traps as |state.displayCards| 1`] = `
Array [
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "trap",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584276957/lightning_ygo_api/card_images/art/91989718.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584276954/lightning_ygo_api/card_images/big/91989718.jpg",
      "id": "91989718",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584276956/lightning_ygo_api/card_images/small/91989718.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Attack Reflector Unit",
    "text": "Tribute 1 \\"Cyber Dragon\\". Special Summon 1 \\"Cyber Barrier Dragon\\" from your hand or Deck.",
    "types": Array [
      "normal",
    ],
  },
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "trap",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280023/lightning_ygo_api/card_images/art/12670770.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280019/lightning_ygo_api/card_images/big/12670770.jpg",
      "id": "12670770",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280022/lightning_ygo_api/card_images/small/12670770.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Cyber Network",
    "text": "Destroy this card during your 3rd Standby Phase after activation. Once per turn, if \\"Cyber Dragon\\" is on the field: You can banish 1 LIGHT Machine monster from your Deck. When this card is sent from the field to the GY: Special Summon as many of your banished LIGHT Machine monsters as possible, and if you do, destroy all Spells and Traps you control. Monsters that were Special Summoned by this effect cannot activate their effects. You cannot conduct your Battle Phase the turn you activate this effect.",
    "types": Array [
      "continuous",
    ],
  },
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "spell",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280052/lightning_ygo_api/card_images/art/86686671.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280049/lightning_ygo_api/card_images/big/86686671.jpg",
      "id": "86686671",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584280051/lightning_ygo_api/card_images/small/86686671.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Cyber Repair Plant",
    "text": "If \\"Cyber Dragon\\" is in your GY: You can activate 1 of these effects. If you have 3 or more \\"Cyber Dragon\\" in your GY at activation, you can activate both effects and resolve in sequence. You can only activate 1 \\"Cyber Repair Plant\\" per turn.
● Add 1 LIGHT Machine monster from your Deck to your hand.
● Target 1 LIGHT Machine monster in your GY; shuffle that target into your Deck.",
    "types": Array [
      "normal",
    ],
  },
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "spell",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942151/lightning_ygo_api/card_images/art/55704856.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942149/lightning_ygo_api/card_images/big/55704856.jpg",
      "id": "55704856",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942150/lightning_ygo_api/card_images/small/55704856.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Cyberload Fusion",
    "text": "Fusion Summon 1 Fusion Monster from your Extra Deck that lists a \\"Cyber Dragon\\" monster as material, by shuffling the Fusion Materials listed on it into the Deck, from among your cards on the field and/or your face-up banished cards, but monsters you control cannot attack for the rest of this turn, except that Fusion Summoned monster. You can only activate 1 \\"Cyberload Fusion\\" per turn.",
    "types": Array [
      "quick-play",
    ],
  },
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "trap",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942154/lightning_ygo_api/card_images/art/82428674.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942152/lightning_ygo_api/card_images/big/82428674.jpg",
      "id": "82428674",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1586942153/lightning_ygo_api/card_images/small/82428674.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Cybernetic Overflow",
    "text": "Banish \\"Cyber Dragon(s)\\" with different Levels from your hand, face-up field, and/or GY, then destroy an equal number of cards your opponent controls. If this card on the field is destroyed by card effect: Add 1 \\"Cyber\\" Spell/Trap from your Deck to your hand. You can only use each effect of \\"Cybernetic Overflow\\" once per turn.",
    "types": Array [
      "normal",
    ],
  },
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "spell",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584282665/lightning_ygo_api/card_images/art/52875873.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584282663/lightning_ygo_api/card_images/big/52875873.jpg",
      "id": "52875873",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584282664/lightning_ygo_api/card_images/small/52875873.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Evolution Burst",
    "text": "If you control \\"Cyber Dragon\\": Target 1 card your opponent controls; destroy that target. \\"Cyber Dragon\\" cannot attack the turn you activate this card.",
    "types": Array [
      "normal",
    ],
  },
  Object {
    "atk": null,
    "attribute": null,
    "cardType": "spell",
    "def": null,
    "image": Object {
      "art": "https://res.cloudinary.com/georgeletsos/image/upload/v1584290469/lightning_ygo_api/card_images/art/66607691.jpg",
      "big": "https://res.cloudinary.com/georgeletsos/image/upload/v1584290466/lightning_ygo_api/card_images/big/66607691.jpg",
      "id": "66607691",
      "small": "https://res.cloudinary.com/georgeletsos/image/upload/v1584290468/lightning_ygo_api/card_images/small/66607691.jpg",
    },
    "level": null,
    "monsterType": null,
    "name": "Photon Generator Unit",
    "text": "Tribute 2 \\"Cyber Dragons\\". Special Summon 1 \\"Cyber Laser Dragon\\" from your hand, Deck, or Graveyard.",
    "types": Array [
      "quick-play",
    ],
  },
]
`;
